mlflow/tracing/python_api.py

```diff
@@ -122,7 +127,58 @@ def start_span(
             mlflow_span = NoOpMLflowSpanWrapper()
             yield mlflow_span
     finally:
-        mlflow_span._end()
+        mlflow_span.end()
+
+
+def start_detached_span(
+    name: str = "span",
+    span_type: Optional[str] = None,
+    parent: Optional[MLflowSpanWrapper] = None,
+) -> MLflowSpanWrapper:
+    """
+    Create a new span and start it without attaching it to the global trace context.
+
+    This is an imperative API to manually create a new span under a specific trace id and parent
+    span, unlike the higher-level APIs like `@mlflow.trace()` and `with mlflow.start_span()`,
+    which automatically manage the span lifecycle and parent-child relationship.
+
+    This API is particularly useful for the case where the automatic context management is not
+    sufficient, such as callback-based instrumentation where span start and end are not in the same
+    call stack, or multi-threaded applications where the context is not propagated automatically.
+
```

Comment:
Should we also mention / indicate that this isn't an 'all or nothing' API and that, provided that the parent_id from the yielded span created in the fluent API can be used to provide the appropriate parent designation to this API?